### PR TITLE
Add Deep delete options for Disks and NICs

### PR DESCRIPTION
**Pull request by basharg**
## Description

Add deep delete options for VM Disks and NICs based on design spec https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/936

Add networkApiVersion for VMSS based on design spec https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/935

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/AddAzureRmVmssNetworkInterfaceConfigurationCommand.cs [hunk 1]
```diff
@@ -106,6 +111,8 @@ private void Run()
                 this.VirtualMachineScaleSet.VirtualMachineProfile.NetworkProfile = new VirtualMachineScaleSetNetworkProfile();
             }
 
+            this.VirtualMachineScaleSet.VirtualMachineProfile.NetworkProfile.NetworkApiVersion = this.NetworkApiVersion?? Microsoft.Azure.Management.Compute.Models.NetworkApiVersion.TwoZeroTwoZeroHyphenMinusOneOneHyphenMinusZeroOne;
```
Comment (szimmerMicrosoft) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/AddAzureRmVmssNetworkInterfaceConfigurationCommand.cs:114]: If there is a networkApiVersion already on the NetworkProfile and this.NetworkApiVersion == null, we should preserve the existing networkApiVersion.

Overriding it with the hardcoded version seems more user hostile than helpful.

File: src/Compute/Compute/VirtualMachine/Config/AddAzureVMNetworkInterfaceCommand.cs [hunk 2]
```diff
@@ -148,12 +162,14 @@ public override void ExecuteCmdlet()
...
                                 Id = nic.Id,
-                                Primary = nic.Primary
+                                Primary = nic.Primary,
+                                DeleteOption = this.IsParameterBound(c => c.DeleteOption) ? this.DeleteOption : null
                             });
                     }
                     else
                     {
                         existingNic.Primary = nic.Primary;
+                        existingNic.DeleteOption = this.IsParameterBound(c => c.DeleteOption) ? this.DeleteOption : null;
```
Comment (szimmerMicrosoft) [inline src/Compute/Compute/VirtualMachine/Config/AddAzureVMNetworkInterfaceCommand.cs:172]: Shouldn't the ternary expression be

existingNic.DeleteOption = this.IsParameterBound(c => c.DeleteOption) ? this.DeleteOption : nic.DeleteOption

**Comment by msJinLei** [comment]
The PR depends on  #15172

**Comment by msJinLei** [comment]
LGTM

**Review by msJinLei — APPROVED** [review]

**Pull request merged**